staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -108,8 +108,11 @@ func GetInstanceProviderID(ctx context.Context, cloud Interface, nodeName types.
 		if err == NotImplemented {
 			return "", err
 		}
+		if err == InstanceNotFound {
+			return "", err
```

Comment:
historically `GetInstanceProviderID` never returned InstanceNotFound and we only expected it form InstanceID. I think this change is correct but we should update the comments for GetInstanceProviderID to reflect this change.

Reply:
See https://github.com/kubernetes/cloud-provider/blob/master/cloud.go#L181 for example

Reply:
Before my change `GetInstanceProviderID` was returning a string
```
"failed to get instance ID from cloud provider: instance not found"
```
and with my change it returns
```
"instance not found"
```

Given that there is `InstanceNotFound = errors.New("instance not found")` defined I felt it's a right thing to do. I will update the comments accordingly

Reply:
The change is fine I think, but keep in mind that some callers may be checking for the `InstanceNotFound` error.